suggest/elasticsearch/src/main/java/com/spotify/heroic/suggest/elasticsearch/SuggestBackendKV.java

```diff
@@ -645,204 +656,221 @@ public boolean isReady() {
                 suggest.endObject();
 
                 final String suggestId = seriesId + ":" + Integer.toHexString(e.hashCode());
-                tagSpan.putAttribute("suggestId", AttributeValue.stringAttributeValue(suggestId));
-                final FutureReporter.Context writeContext = reporter.setupWriteReporter();
-
-                final ResolvableFuture<IndexResponse> tagWrite = async.future();
-                c.index(tagIndex, TAG_TYPE)
-                    .setId(suggestId)
-                    .setSource(suggest)
-                    .setOpType(DocWriteRequest.OpType.CREATE)
-                    .execute(bind(tagWrite));
-
-                writes.add(
-                    tagWrite
-                        .directTransform(response -> timer.end())
-                        .catchFailed(
-                            handleVersionConflict(
-                                WriteSuggest::new, reporter::reportWriteDroppedByDuplicate))
-                        .onDone(writeContext)
-                        .onFinished(tagSpan::end));
+                final IndexRequestBuilder tagIndexRequest =
+                    c.index(tagIndex, TAG_TYPE)
+                        .setId(suggestId)
+                        .setSource(suggest)
+                        .setOpType(DocWriteRequest.OpType.CREATE);
+                bulkRequestBuilder.add(tagIndexRequest);
               }
             }
-
             indexScope.close();
           }
 
-          rootScope.close();
-          return async
-              .collect(writes, WriteSuggest.reduce())
+          // nothing to write because the series is in the write cache.
+          if (bulkRequestBuilder.numberOfActions() == 0) {
+            return async.resolved(new WriteSuggest());
+          }
+
+          bulkRequestBuilder.execute(bind(future));
+
+          return future
+              .directTransform(
+                  response -> {
+                    boolean addFailureAnnotation = true;
+                    final List<RequestError> errors = new ArrayList<>();
+                    for (final BulkItemResponse r : response.getItems()) {
+                      if (r.isFailed()) {
+                        errors.add(NodeError.internalError(r.getFailureMessage()));
```

Comment:
With the way `RequestError` is designed a `NodeError` here is easiest error to use without modifying the response API's dramatically.